flink-streaming-connectors/flink-connector-cassandra/src/test/java/org/apache/flink/connectors/cassandra/CassandraConnectorTest.java

```
@@ -0,0 +1,307 @@
+
+                    private boolean running = true;
+                    private volatile long cnt = 0;
+
+                    @Override
+                    public void run(SourceContext<Pojo> ctx) throws Exception {
+                        while (running) {
+                            ctx.collect(new Pojo(cnt, "cassandra-" + cnt));
+                            cnt++;
+                            if (cnt == COUNT) {
+                                cancel();
+                            }
+                        }
+                    }
+
+                    @Override
+                    public void cancel() {
+                        running = false;
+                    }
+                });
+
+        source.addSink(new CassandraMapperSink<Pojo>(CREATE_TABLE_MAPPER,Pojo.class) {
+
+            @Override
+            public Builder configureCluster(Builder cluster) {
+                return CassandraConnectorTest.builder;
+            }
+        });
+
+		env.execute();
+
+        ResultSet rs = session.execute(SELECT_QUERY_MAPPER);
+        Assert.assertEquals(COUNT,rs.all().size());
+        session.execute(DROP_TABLE_MAPPER);
+    }
+
+
+	//
+	//	CassandraOutputFormat.java
+	//	CassandraInputFormat.java
+	//
+
+	@Test
+	public void batch(){
+
+		ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
+
+		DataSet<Tuple2<Long,String>> dataSet = env.fromCollection(collection);
+
+		dataSet.output(new CassandraOutputFormat<Tuple2<Long,String>>(CREATE_TABLE,INSERT_QUERY) {
+
+			@Override
+			public Builder configureCluster(Builder cluster) {
+				return builder;
+			}
+		});
+
+		DataSet<Tuple2<Long,String>> inputDS = env.createInput(new CassandraInputFormat<Tuple2<Long,String>>(SELECT_QUERY) {
+
+			@Override
+			public Builder configureCluster(Builder cluster) {
+				return builder;
+			}
+		});
+
+		try {
+			long count =inputDS.count();
+			Assert.assertEquals(count, 20L);
+			session.execute(DROP_TABLE);
+		} catch (Exception e) {
```

you can just throw this exception